server/sonar-search/src/main/java/org/sonar/search/EsSettings.java

```diff
@@ -149,6 +149,31 @@ private InetAddress readHost() {
     }
   }
 
+  private String readInitialStateTimeout() {
+    String initialStateTimeout = props.value(ProcessProperties.SEARCH_INITIAL_STATE_TIMEOUT, "30s").toLowerCase().trim();
+
+    if (initialStateTimeout.endsWith("ms") ||
```

Comment:
No need to validate value. I think propagating the value to Elasticsearch is enough. Otherwise some unit tests should be added :-)